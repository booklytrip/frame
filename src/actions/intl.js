/**
 * @flow
 */

import type { Locale } from '../types';

/**
 * Change current locale to specified
 *
 * @param {String} locale - A new locale to set
 */
export const updateLocale = (locale: Locale) => ({
    type: 'INTL.UPDATE_LOCALE',
    locale,
});
