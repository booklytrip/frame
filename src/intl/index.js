import { addLocaleData } from 'react-intl';
import { reduce, mapValues } from 'lodash';

import en from 'react-intl/locale-data/en';
import lv from 'react-intl/locale-data/lv';
import ru from 'react-intl/locale-data/ru';

import * as defaultLocales from './locales';
import { locales as uikitLocales } from 'uikit';

addLocaleData([
    ...en,
    ...lv,
    ...ru,
]);

/**
 * Combine all messages from list of provides locales
 * and return single object
 */
const combineLocales = (locales) => (
    reduce(locales, (result, locale) => (
        mapValues(locale, (messages, language) => (
            { ...result[language], ...messages }
        ))
    ), {})
);

const locales = combineLocales([defaultLocales, uikitLocales]);

export { locales };
