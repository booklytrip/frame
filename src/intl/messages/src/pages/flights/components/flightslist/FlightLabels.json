[
  {
    "id": "flights.flightslist.BestFlightLabel",
    "defaultMessage": "BEST PRICE / FLIGHT DURATION RATIO"
  },
  {
    "id": "flights.flightslist.DirectFlightLabel",
    "defaultMessage": "CHOOSE DIRECT FLIGHT AND SAVE YOUR TIME!"
  }
]