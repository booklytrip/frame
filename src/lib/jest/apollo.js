import React from 'react';

import { ApolloProvider as Provider } from 'react-apollo';
import ApolloClient from 'apollo-client';
const client = new ApolloClient();

export const ApolloProvider = ({ children }) => (
    <Provider client={ client }>
        { children }
    </Provider>
);
