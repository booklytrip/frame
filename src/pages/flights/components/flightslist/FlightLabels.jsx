/**
 * @flow
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import { darken } from 'polished';
import { get, has } from 'lodash';

import { Icon, Label as BaseLabel } from 'semantic-ui-react';

import type { Flight } from '../../../../types';

type FlightLabelsProps = {
    flight: Flight,
    best: boolean,
};

const Label = styled(BaseLabel)`
    margin-bottom: 1rem !important;
    color: #FFFFFF !important;
`;

const BestFlightLabelWrapper = styled(Label)`
    background-color: ${({ theme }) =>
        theme.flights.bestFlightLabel.background} !important;
    border-color: ${({ theme }) =>
        darken(0.2, theme.flights.bestFlightLabel.background)} !important;
    color: ${({ theme }) => theme.flights.bestFlightLabel.color} !important;
`;

const DirectFlightLabelWrapper = styled(Label)`
    background-color: ${({ theme }) =>
        theme.flights.directFlightLabel.background} !important;
    border-color: ${({ theme }) =>
        darken(0.2, theme.flights.directFlightLabel.background)} !important;
    color: ${({ theme }) => theme.flights.directFlightLabel.color} !important;
`;

/**
 * Verifies flight direct and return true if flight is direct in both sides
 *
 * @param {Object} flight - A flight object
 */
const isDirectFlight = (flight: Flight) => {
    return (
        get(flight, 'forwardSector.stops') === 0 &&
        (!has(flight, 'comebackSector') ||
            get(flight, 'comebackSector.stops') === 0)
    );
};

/**
 * Label for best flight option where ratio of price and duration is optimal
 */
const BestFlightLabel = () =>
    <BestFlightLabelWrapper size="large" ribbon>
        <Icon name="thumbs star" />
        <FormattedMessage
            id="flights.flightslist.BestFlightLabel"
            defaultMessage="BEST PRICE / FLIGHT DURATION RATIO"
        />
    </BestFlightLabelWrapper>;

/**
 * Label for flight with direct direction in both sides
 */
const DirectFlightLabel = () =>
    <DirectFlightLabelWrapper size="large" ribbon>
        <Icon name="thumbs up" />
        <FormattedMessage
            id="flights.flightslist.DirectFlightLabel"
            defaultMessage="CHOOSE DIRECT FLIGHT AND SAVE YOUR TIME!"
        />
    </DirectFlightLabelWrapper>;

/**
 * Shows best label for specified flight
 *
 * @param {Object} flight - A flight object
 * @param {Bool}   best   - A flight is best option
 */
const FlightLabels = ({ flight, best }: FlightLabelsProps) => {
    if (best) {
        return <BestFlightLabel />;
    }

    if (isDirectFlight(flight)) {
        return <DirectFlightLabel />;
    }

    return null;
};

export { BestFlightLabel, DirectFlightLabel };

export default FlightLabels;
