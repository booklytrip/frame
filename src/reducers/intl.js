/**
 * @flow
 */

import { Map } from 'immutable';
import { saveState, loadState } from '../lib/sessionStorage';

const initialState = new Map(loadState('intl') || {});

export default function(state: Object = initialState, action: Object) {
    switch (action.type) {
        // Replace current locale with specified
        case 'INTL.UPDATE_LOCALE': {
            const newState = state.set('locale', action.locale);
            saveState('intl', newState);
            return newState;
        }
        default:
            return state;
    }
}
