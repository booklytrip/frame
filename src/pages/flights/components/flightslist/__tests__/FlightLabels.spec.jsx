import React from 'react';
import { createWithIntl } from '../../../../../lib/jest/intl';
import FlightLabels from '../FlightLabels';

const directFlight = {
    forwardSector: {
        stops: 0,
    },
};

describe('FlightLabels', () => {
    describe('Snapshots', () => {
        it('renders best flight label', () => {
            const rendered = createWithIntl(
                <FlightLabels best />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });

        it('renders direct flight label', () => {
            const rendered = createWithIntl(
                <FlightLabels flight={ directFlight } />
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    })
});
