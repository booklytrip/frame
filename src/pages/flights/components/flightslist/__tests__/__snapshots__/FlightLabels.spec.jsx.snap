// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightLabels Snapshots renders best flight label 1`] = `
<div
  className="ui yellow large ribbon label SCbVL"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="thumbs outline up icon"
  />
  <span>
    BEST PRICE / FLIGHT DURATION RATIO
  </span>
</div>
`;

exports[`FlightLabels Snapshots renders direct flight label 1`] = `
<div
  className="ui blue large ribbon label SCbVL"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="thumbs outline up icon"
  />
  <span>
    CHOOSE DIRECT FLIGHT AND SAVE YOUR TIME!
  </span>
</div>
`;
